File: LeetCode/Easy/Count Pairs Whose Sum is Less than Target/2824.java
class Solution {
    public int countPairs(List<Integer> nums, int target) {
        int count = 0;
        int n = nums.size();

        Collections.sort(nums);

        int left = 0, right = n - 1;
        while (left < right) {
            if (nums.get(left) + nums.get(right) < target) {
                count += (right - left);
                left++;
            } else {
                right--;
            }
        }

        return count;
    }
}

File: LeetCode/Easy/Count Pairs Whose Sum is Less than Target/2824.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var countPairs = function(nums, target) {
    let count = 0;
    const n = nums.length;

    nums.sort((a, b) => a - b);

    let left = 0, right = n - 1;
    while (left < right) {
        if (nums[left] + nums[right] < target) {
            count += (right - left);
            left++;
        } else {
            right--;
        }
    }

    return count;
};
